include/odp/api/spec/ipsec.h

```diff
@@ -272,6 +272,17 @@ typedef struct odp_ipsec_capability_t {
 	 */
 	odp_support_t inline_ipsec_tm;
 
+	/** Insert outer header as part of inline outbound operation.
+	 *
+	 * * 0: Application is required to update packet with outer headers
+	 *      such that packet is ready for transmission after successful
+	 *      IPSEC transformation. IPSEC transformation would update
+	 *      fields starting from L3 offset.
+	 * * 1: Application would provide outer header as part of params to
+	 *      odp_ipsec_out_inline() API. (default)
+	 */
+	odp_bool_t outer_hdr_insert;
```

Comment:
Instead of even having capability, changing the semantics  always for odp_ipsec_out_inline() make sure l2 data is intact in packet, is better for applications as otherwise, applications need to check for this capability (atleast upstream examples) which impacts fast path.

Reply:
L2 data immediately preceding L3 may not be true for all platform implementations. For example, for purely s/w implementations, the IPsec tunnel header additions etc would mean packet would grow into headroom, anticipating which would be very cumbersome for the application.

I had described this in the commit description of the API change also. Please take a look at it.